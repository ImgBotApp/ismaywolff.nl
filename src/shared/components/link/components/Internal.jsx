import { Link as ReactRouterLink } from 'react-router-dom'
import { string, bool } from 'prop-types'
import styled from 'styled-components'

const setBackgroundImage = props => {
  if (props.clean) {
    return 'unset'
  }

  return 'linear-gradient(transparent 50%, currentColor 50%)'
}

const Internal = styled(ReactRouterLink)`
  background-image: ${setBackgroundImage};
  background-position: 0 1.07em;
  background-repeat: repeat-x;
  background-size: 2px 0.1em;
  color: currentColor;
  font-size: ${props => props.size};
  text-decoration: none;
`

Internal.propTypes = {
  to: string.isRequired,
  clean: bool,
  size: string
}

Internal.defaultProps = {
  clean: false,
  size: 'unset'
}

export default Internal
