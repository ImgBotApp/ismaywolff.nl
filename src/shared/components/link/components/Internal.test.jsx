import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Internal from './Internal'

describe('<Internal />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Internal to="/url">Text</Internal>)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders clean correctly', () => {
    const wrapper = shallow(
      <Internal to="/url" clean>
        Text
      </Internal>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders size correctly', () => {
    const wrapper = shallow(
      <Internal to="/url" size="16px">
        Text
      </Internal>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
