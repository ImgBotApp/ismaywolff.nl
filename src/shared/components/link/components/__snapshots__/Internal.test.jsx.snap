// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Internal /> renders clean correctly 1`] = `
.c0 {
  background-image: unset;
  background-position: 0 1.07em;
  background-repeat: repeat-x;
  background-size: 2px 0.1em;
  color: currentColor;
  font-size: unset;
  text-decoration: none;
}

<Link
  className="c0"
  clean={true}
  replace={false}
  size="unset"
  to="/url"
>
  Text
</Link>
`;

exports[`<Internal /> renders correctly 1`] = `
.c0 {
  background-image: linear-gradient(transparent 50%, currentColor 50%);
  background-position: 0 1.07em;
  background-repeat: repeat-x;
  background-size: 2px 0.1em;
  color: currentColor;
  font-size: unset;
  text-decoration: none;
}

<Link
  className="c0"
  clean={false}
  replace={false}
  size="unset"
  to="/url"
>
  Text
</Link>
`;

exports[`<Internal /> renders size correctly 1`] = `
.c0 {
  background-image: linear-gradient(transparent 50%, currentColor 50%);
  background-position: 0 1.07em;
  background-repeat: repeat-x;
  background-size: 2px 0.1em;
  color: currentColor;
  font-size: 16px;
  text-decoration: none;
}

<Link
  className="c0"
  clean={false}
  replace={false}
  size="16px"
  to="/url"
>
  Text
</Link>
`;
